app/services/DeskproSubmission.scala

```diff
@@ -129,10 +129,13 @@ object DeskproSubmission {
   def replaceReferrerPath(referrer: String, path: Option[String]): String =
     path
       .filter(_.trim.nonEmpty)
-      .map(p => buildUri(referrer).setPath(p).build().toASCIIString)
+      .map { p =>
+        val userActionUrlPath = if (p.startsWith("/")) p else s"/$p"
+        buildUri(referrer).resolve(userActionUrlPath).toASCIIString
+      }
```

Comment:
we've used the name "userActionUrlPath" within this utility method while userAction is an external thing, what about using "absolutePath"? Though somewhat feels like if we want absolute paths that we should be requiring them rather than taking a relative and treating it like an absolute one - but if it's how it works at the moment in practice if someone gives us a relative path then that makes sense 🤷‍♂️

Reply:
I did struggle to name this, i like absolutePath as it describes the usage well. Ill change it

Reply:
also, just to be cheeky and ask rather than checking, there's test coverage for this that proves it behaves the same after as it did before?

Reply:
Yes, this [spec](https://github.com/hmrc/contact-frontend/blob/main/test/services/DeskproSubmissionSpec.scala) has all the current tests around building a URI